### PR TITLE
Adding screen share support to live_api_starter.py

### PR DESCRIPTION
**Pull request by clementfarabet**

### CODE REVIEW

**Review by logankilpatrick — DISMISSED** [review]

File: gemini-2/live_api_starter.py [hunk 1]
```diff
@@ -113,6 +117,33 @@ async def get_frames(self):
         # Release the VideoCapture object
         cap.release()
 
+    def _get_screen(self, sct, monitor):
```
Comment (rahmatspaucs6) [inline gemini-2/live_api_starter.py:120]: Sreen black and text green

File: gemini-2/live_api_starter.py [hunk 2]
```diff
@@ -113,6 +117,33 @@ async def get_frames(self):
...
+        i = sct.grab(monitor)
```
Comment (rahmatspaucs6) [inline gemini-2/live_api_starter.py:121]: /010101

**Comment by clementfarabet** [comment]
Just added argparse to make it easier to switch from camera to screen

File: gemini-2/live_api_starter.py (hunk 1 above)
Comment (Giom-V) [reply gemini-2/live_api_starter.py:125]: Likely an AI bot

File: gemini-2/live_api_starter.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 
 # To install the dependencies for this script, run:
-#  pip install google-genai opencv-python pyaudio pillow
+#  pip install google-genai opencv-python pyaudio pillow mss
 # And to run this script, ensure the GOOGLE_API_KEY environment
 # variable is set to the key you obtained from Google AI Studio.
```
Comment (Giom-V) [inline gemini-2/live_api_starter.py:19]: I think we should also add a line on how to enable screensharing vs. camera

**Comment by Giom-V** [comment]
We will also need to update the websocket version: https://github.com/google-gemini/cookbook/blob/main/gemini-2/websockets/live_api_starter.py
I can do it myself if you don't have the time.

File: gemini-2/live_api_starter.py [hunk 4]
```diff
@@ -113,6 +122,33 @@ async def get_frames(self):
...
+
+        image_bytes = image_io.read()
+        return {"mime_type": mime_type, "data": base64.b64encode(image_bytes).decode()}
+
+    async def get_screen(self):
+        sct = mss.mss()
+        monitor = sct.monitors[0]
+
+        while True:
+            frame = await asyncio.to_thread(self._get_screen, sct, monitor)
```
Comment (Giom-V) [inline gemini-2/live_api_starter.py:144]: @clementfarabet I'm getting that error when running on my windows PC:
| Traceback (most recent call last):
| File "C:\Users\giom\myenv_test\live_api_starter.py", line 146, in get_screen
| frame = await asyncio.to_thread(self._get_screen, sct, monitor)
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
| File "C:\Users\giom\AppData\Local\Programs\Python\Python312\Lib\asyncio\threads.py", line 25, in to_thread
| return await loop.run_in_executor(None, func_call)
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
| File "C:\Users\giom\AppData\Local\Programs\Python\Python312\Lib\concurrent\futures\thread.py", line 58, in run
| result = self.fn(*self.args, **self.kwargs)
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
| File "C:\Users\giom\myenv_test\live_api_starter.py", line 128, in _get_screen
| i = sct.grab(monitor)
| ^^^^^^^^^^^^^^^^^
| File "C:\Users\giom\myenv_test\Lib\site-packages\mss\base.py", line 101, in grab
| screenshot = self._grab_impl(monitor)
| ^^^^^^^^^^^^^^^^^^^^^^^^
| File "C:\Users\giom\myenv_test\Lib\site-packages\mss\windows.py", line 226, in _grab_impl
| srcdc, memdc = self._handles.srcdc, self._handles.memdc
| ^^^^^^^^^^^^^^^^^^^
| AttributeError: '_thread._local' object has no attribute 'srcdc'

Any idea what the cause could be?

Comment (Giom-V) [reply gemini-2/live_api_starter.py:144]: This could be the reason: https://stackoverflow.com/a/76600858

File: gemini-2/live_api_starter.py (hunk 3 above)
Comment (clementfarabet) [reply gemini-2/live_api_starter.py:19]: +1, added as a second commit https://github.com/google-gemini/cookbook/pull/359/commits/dacb16b34ad50c06df847e8eead68ee4b54e54ea

File: gemini-2/live_api_starter.py (hunk 4 above)
Comment (clementfarabet) [reply gemini-2/live_api_starter.py]: With this fix https://github.com/google-gemini/cookbook/pull/359/commits/44c4a95433691d9e0b4457b7abdf8c7dcdf0335e - works?

**Review by Giom-V — APPROVED** [review]

**Pull request merged**